website/playground/panels.jsx

```diff
@@ -1,17 +1,116 @@
-import { onMounted, onUnmounted, useTemplateRef, watch } from "vue";
-
-const { CodeMirror } = window;
+import { catppuccinLatte, catppuccinMocha } from "@catppuccin/codemirror";
+import { closeBrackets, closeBracketsKeymap } from "@codemirror/autocomplete";
+import {
+  defaultKeymap,
+  history,
+  historyKeymap,
+  indentWithTab,
+} from "@codemirror/commands";
+import {
+  bracketMatching,
+  foldable,
+  foldEffect,
+  foldGutter as foldGutterExt,
+  foldKeymap,
+} from "@codemirror/language";
+import {
+  Compartment,
+  EditorState,
+  StateEffect,
+  StateField,
+} from "@codemirror/state";
+import {
+  Decoration,
+  EditorView,
+  keymap,
+  lineNumbers as lineNumbersExt,
+} from "@codemirror/view";
+import { inject, onMounted, onUnmounted, useTemplateRef, watch } from "vue";
+import { useTheme } from "./composables/use-theme.js";
 
 function setup(props, { emit }) {
-  const textareaRef = useTemplateRef("textarea");
+  const editorRef = useTemplateRef("editorRef");
+
   let _codeMirror = null;
   let _cached = "";
-  let _overlay = null;
+  const _overlay = StateEffect.define();
+
+  const overlayField = StateField.define({
+    create() {
+      return Decoration.none;
+    },
+    update(decorations, tr) {
+      for (const effect of tr.effects) {
+        if (effect.is(_overlay)) {
+          const { start, end } = effect.value;
+          if (start !== undefined && end !== undefined && start < end) {
+            return Decoration.set([overlayMark.range(start, end)]);
+          }
+          return Decoration.none;
+        }
+      }
+      return decorations.map(tr.changes);
+    },
+    provide: (f) => EditorView.decorations.from(f),
+  });
+
+  const overlayMark = Decoration.mark({
+    class: "cm-overlay-highlight",
+  });
+
+  const languageExt = new Compartment();
+  const rulerTheme = new Compartment();
+  const foldGutterCompartment = new Compartment();
+
+  const { theme } = useTheme();
+  const editorTheme = new Compartment();
+
   const onChange = (value) => {
     emit("change", value);
   };
 
-  const componentDidMount = () => {
+  const createRulerTheme = (printWidth, color) => {
+    if (!printWidth) {
+      return EditorView.theme({});
+    }
+
+    return EditorView.theme({
+      ".cm-content::before": {
+        content: '""',
+        position: "absolute",
+        top: "0",
+        bottom: "0",
+        left: `${printWidth}ch`,
```

Comment:
Ha, style is incorrect.

Reply:
Fixed [`9350331` (#18524)](https://github.com/prettier/prettier/pull/18524/commits/9350331d394e5206f6cc2f1db09bbee6ff858d68)